modin/core/execution/ray/common/utils.py

```diff
@@ -187,18 +191,11 @@ def _get_object_store_memory() -> Optional[int]:
     )
     if object_store_memory == 0:
         return None
-    # Versions of ray with MAC_DEGRADED_PERF_MMAP_SIZE_LIMIT don't allow us
-    # to initialize ray with object store size larger than that constant.
+    # Newer versions of ray don't allow us to initialize ray with object store
+    # size larger than that _MAC_OBJECT_STORE_LIMIT_BYTES.
     # For background see https://github.com/ray-project/ray/issues/20388
-    mac_size_limit = getattr(
-        ray.ray_constants, "MAC_DEGRADED_PERF_MMAP_SIZE_LIMIT", None
-    )
-    if (
-        sys.platform == "darwin"
-        and mac_size_limit is not None
-        and object_store_memory > mac_size_limit
-    ):
-        object_store_memory = mac_size_limit
+    if sys.platform == "darwin":
+        object_store_memory = min(object_store_memory, _MAC_OBJECT_STORE_LIMIT_BYTES)
```

Comment:
Doesn't this mean we'll be limiting the object store memory even with older versions of Ray that may not have this bug?

Or is this a Mac issue that affects all versions of Ray?

If it's not the former, and it is version dependent (either on the arch version or ray version) perhaps it makes sense to check before limiting the object store memory?

Reply:
I wrote a comment on the issue [here](https://github.com/modin-project/modin/issues/4872#issuecomment-1225238172) with what I've found from digging into this bug. In conclusion, I think we should apply the limit to ray >= 1.3, so that's what I've done.

Reply:
That sounds good to me!